blob/singularity_store.go

```diff
@@ -0,0 +1,85 @@
+package blob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/data-preservation-programs/singularity/model"
+	"github.com/filecoin-project/motion/singularity"
+)
+
+type SingularityService interface {
+	Start(ctx context.Context, dir string) error
+	GetItem(ctx context.Context, id uint64) (*model.Item, error)
+	PushItem(ctx context.Context, path string) (*model.Item, error)
+}
+
+type SingularityStore struct {
+	local       *LocalStore
+	singularity SingularityService
+}
+
+func OpenSingularityStore(dir string) (*SingularityStore, error) {
+	dbFile := path.Join(path.Clean(dir), "singularity.db")
+	singularityService, err := singularity.OpenSQLiteSingularity(dbFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open singularity database: %w", err)
+	}
+	local := NewLocalStore(dir)
+	return NewSingularityStore(local, singularityService), nil
+}
+
+func NewSingularityStore(local *LocalStore, singularity SingularityService) *SingularityStore {
+	return &SingularityStore{
+		local:       local,
+		singularity: singularity,
+	}
+}
+
+func (l *SingularityStore) Start(ctx context.Context) error {
+	return l.singularity.Start(ctx, l.local.dir)
```

Comment:
Looks like this is the only functional call that changed the blobstore interface to add `Start` and `Shutdown`? My vote would be to keep the blob.Store interface as small as possible.

Overall, my plan re Start Shutdown was to introduce a service interface and type check in `Motion` to start stop store if it is a service since there will be many more such service components that `Motion` would need to start stop.

Reply:
I did it cause of the Start/Shutdown also being in the RIBS interface. I agree though, let's keep it small.